src/components/outer-link.tsx

```diff
@@ -0,0 +1,43 @@
+"use client";
+
+import React, { useMemo } from "react";
+
+import { ExternalLink } from "lucide-react";
+
+import InstagramLogo from "../../public/Instagram.svg";
+import SNSLogo from "../../public/SNS.svg";
+import TwitterLogo from "../../public/Twitter.svg";
+import { POPULAR_SNS } from "../constants/constants";
+
+type Props = {
+  siteName: string;
+  siteUrl: string;
+};
+
+const OuterLink = ({ siteName, siteUrl }: Props) => {
+  const includeTwitter = siteUrl.includes(POPULAR_SNS.TWITTER);
+  const includeInstagram = siteUrl.includes(POPULAR_SNS.INSTAGRAM);
+  const unincludePopularSns = useMemo(() => {
+    if (includeTwitter || includeInstagram) {
+      return false;
+    } else {
+      return true;
+    }
+  }, [includeTwitter, includeInstagram]);
+  return (
+    <div className="flex items-center h-extend-22 gap-x-4 p-4 border-b-2">
+      {includeTwitter && <TwitterLogo className="rounded-2xl" width={56} height={56} />}
+      {includeInstagram && <InstagramLogo className="rounded-2xl" width={56} height={56} />}
+      {unincludePopularSns && <SNSLogo className="rounded-2xl" width={56} height={56} />}
+      <div className="grid">
+        <p className="text-base">{siteName}</p>
+        <p className="text-sm text-mauve6">{siteUrl}</p>
+      </div>
+      <a href={siteUrl} target="_blank" rel="noopener noreferrer">
+        <ExternalLink size={24} />
+      </a>
+    </div>
```

Comment:
> 自身が対応した箇所は約400pxほどであり、それを元にFigmaも作成されているようなので、そこらへんの細かい対応は別途行う予定。

と書かれていますが、以下のように最低限のレイアウトは対応しておいても良いのかなと思いました！

![スクリーンショット 2023-06-25 9 23 52](https://github.com/qin-team-recipe/12-recipe-app/assets/63396451/074d0ca1-e5a1-4adf-be30-a2af526e5660)


```suggestion
    <div className="flex items-center justify-between gap-x-4 p-4 border-b-2">
      {includeTwitter && <TwitterLogo className="rounded-2xl" width={56} height={56} />}
      {includeInstagram && <InstagramLogo className="rounded-2xl" width={56} height={56} />}
      {includedPopularSns && <SNSLogo className="rounded-2xl" width={56} height={56} />}
      <div className="flex flex-col flex-1">
        <p className="text-mauve12">{siteName}</p>
        <p className="text-sm text-mauve11">{siteUrl}</p>
      </div>
      <a href={siteUrl} target="_blank" rel="noopener noreferrer">
        <ExternalLink size={24} className="text-mauve11" />
      </a>
    </div>
```

Reply:
ありがとうございます！
修正しておきます。